src/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.WriteProperties.Decimal.cs

```diff
@@ -0,0 +1,259 @@
+        /// </exception>
+        public void WriteNumber(string propertyName, decimal value, bool suppressEscaping = false)
+            => WriteNumber(propertyName.AsSpan(), value, suppressEscaping);
+
+        /// <summary>
+        /// Writes the property name and <see cref="decimal"/> value (as a JSON number) as part of a name/value pair of a JSON object.
+        /// </summary>
+        /// <param name="propertyName">The UTF-16 encoded property name of the JSON object to be transcoded and written as UTF-8.</param>
+        /// <param name="value">The value to be written as a JSON number as part of the name/value pair.</param>
+        /// <param name="suppressEscaping">If this is set, the writer assumes the property name is properly escaped and skips the escaping step.</param>
+        /// <exception cref="ArgumentException">
+        /// Thrown when the specified property name is too large.
+        /// </exception>
+        /// <exception cref="JsonWriterException">
+        /// Thrown if this would result in an invalid JSON to be written (while validation is enabled).
+        /// </exception>
+        public void WriteNumber(ReadOnlySpan<char> propertyName, decimal value, bool suppressEscaping = false)
+        {
+            JsonWriterHelper.ValidateProperty(ref propertyName);
+
+            if (!suppressEscaping)
+                WriteNumberSuppressFalse(ref propertyName, value);
+            else
+                WriteNumberByOptions(ref propertyName, value);
+
+            _currentDepth |= 1 << 31;
+            _tokenType = JsonTokenType.Number;
+        }
+
+        /// <summary>
+        /// Writes the property name and <see cref="decimal"/> value (as a JSON number) as part of a name/value pair of a JSON object.
+        /// </summary>
+        /// <param name="propertyName">The UTF-8 encoded property name of the JSON object to be written.</param>
+        /// <param name="value">The value to be written as a JSON number as part of the name/value pair.</param>
+        /// <param name="suppressEscaping">If this is set, the writer assumes the property name is properly escaped and skips the escaping step.</param>
+        /// <exception cref="ArgumentException">
+        /// Thrown when the specified property name is too large.
+        /// </exception>
+        /// <exception cref="JsonWriterException">
+        /// Thrown if this would result in an invalid JSON to be written (while validation is enabled).
+        /// </exception>
+        public void WriteNumber(ReadOnlySpan<byte> propertyName, decimal value, bool suppressEscaping = false)
+        {
+            JsonWriterHelper.ValidateProperty(ref propertyName);
+
+            if (!suppressEscaping)
+                WriteNumberSuppressFalse(ref propertyName, value);
+            else
+                WriteNumberByOptions(ref propertyName, value);
+
+            _currentDepth |= 1 << 31;
```

Comment:
This line occurs a lot, and it's not clear at the call site why it's being done.  Put it into a method that's appropriately named?